src/ucp/core/ucp_ep.c

```diff
@@ -1143,10 +1144,38 @@ static void ucp_ep_set_lanes_failed(ucp_ep_h ep, uct_ep_h *uct_eps)
     }
 }
 
+static int ucp_ep_can_unprogress_lane(ucp_ep_h ep, uct_ep_h uct_ep)
+{
+    if (!ep->worker->context->config.ext.adaptive_progress) {
+        return 0;
+    }
+
+    if (!ucp_ep_has_cm_lane(ep)) {
```

Comment:
why needed? we should unprogress also direct connection endpoints

Reply:
what about one sided remotely connected endpoints? can we wrongly unprogress iface then?

Reply:
we progress both one-sided ep: https://github.com/openucx/ucx/blob/de6bcbdeafe6160d0926d80629cc0a14e831c677/src/ucp/wireup/wireup.c#L938
and 2-sided ep: 
https://github.com/openucx/ucx/blob/de6bcbdeafe6160d0926d80629cc0a14e831c677/src/ucp/wireup/wireup.c#L978

Reply:
> we progress both one-sided ep:

IIUC that's for locally created EPs... BTW can we have truly one-sided connected EPs? or we always create internal EP for REQ_ACK?

Reply:
we can have one-sided DC ep, if we use it only for small messages/hw rma (do not resolve remote ep id) then only sender side should call "worker_iface_progress_ep"

Reply:
and then TL adds its progress by solicited event without incrementing counter in `ucp_worker_iface_progress_ep`?

Reply:
Yes, on recv side it's done here:
https://github.com/openucx/ucx/blob/1a620d38a8476f59d34e33468199340e310e5785/src/ucp/core/ucp_worker.c#L611

Reply:
ok,  `wiface->activate_count` will be incremented in `ucp_worker_iface_activate` there so it cannot be deactivated by "ep_unprogress", then `ucp_ep_has_cm_lane` condition can be removed. thanks for clarification!